### PR TITLE
Update dskit.

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR updates dskit to fetch latest changes:
* https://github.com/grafana/dskit/pull/564 (no user-visible changes)
* https://github.com/grafana/dskit/pull/566 (fixes bug introduced earlier this week)

#### Checklist

- [na] Tests updated.
- [na] Documentation added.
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [na] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by aldernero — APPROVED** [review]
LGTM

**Comment by pstibrany** [comment]
There was a failure in `TestOwnedSeriesServiceWithIngesterRing` test https://github.com/grafana/mimir/actions/runs/10422335813/job/28866598663. I've rerun the test locally with `-count=10` (it takes ~5m) and can't reproduce the failure. Given the test uses time-based polling quite a bit, I suspect failure was due to CI machine slowness.

**Comment by pstibrany** [comment]
> There was a failure in `TestOwnedSeriesServiceWithIngesterRing` test https://github.com/grafana/mimir/actions/runs/10422335813/job/28866598663. I've rerun the test locally with `-count=10` (it takes ~5m) and can't reproduce the failure. Given the test uses time-based polling quite a bit, I suspect failure was due to CI machine slowness.

Unfortunately this failed again, with exact same errors. I don't think it's a timing issue. Looking at the code and trying to troubleshoot it locally was not successful yet.

File: pkg/ingester/owned_series.go [hunk 1]
```diff
@@ -290,14 +290,22 @@ func (ir *ownedSeriesIngesterRingStrategy) shardSizeForUser(userID string) int {
 }
 
 func (ir *ownedSeriesIngesterRingStrategy) tokenRangesForUser(userID string, shardSize int) (ring.TokenRanges, error) {
+	all, _ := ir.ingestersRing.GetAllHealthy(ring.Read)
```
Comment (pracucci) [inline pkg/ingester/owned_series.go:293]: Looks like some debug code has been unintentionally committed.

File: pkg/ingester/owned_series_test.go [hunk 2]
```diff
@@ -77,6 +77,7 @@ func (c *ownedSeriesTestContextBase) checkActiveSeriesCount(t *testing.T, expect
 func (c *ownedSeriesTestContextBase) updateOwnedSeriesAndCheckResult(t *testing.T, ringChanged bool, expectedUpdatedTenants int, expectedReason string) {
 	c.buf.Reset()
 	require.Equal(t, expectedUpdatedTenants, c.ownedSeries.updateAllTenants(context.Background(), ringChanged), c.buf.String())
+	t.Log(c.buf.String())
```
Comment (pracucci) [inline pkg/ingester/owned_series_test.go:80]: Is this change intentional too?

**Comment by pracucci** [comment]
I left some comments about debug code in this PR, but I haven't realised you are debugging it in CI. I removed by comments.

File: pkg/ingester/owned_series_test.go (hunk 2 above)
Comment (pstibrany) [inline pkg/ingester/owned_series_test.go:80]: I'd keep this. It will only log when tests fail.

**Comment by pstibrany** [comment]
Fix for test failures: https://github.com/grafana/dskit/pull/567
